implementations/rust/ockam/ockam_command/src/message/send.rs

```diff
@@ -91,13 +95,40 @@ async fn rpc(mut ctx: Context, (opts, cmd): (CommandGlobalOpts, SendCommand)) ->
         let mut rpc = RpcBuilder::new(ctx, opts, &api_node)
             .tcp(tcp.as_ref())?
             .build();
-        rpc.request_with_timeout(req(&to, &cmd.message), Duration::from_secs(cmd.timeout))
+
+        // must bind hex decoding vec to later get its slice
+        let hex_decode_result = hex::decode(cmd.message.to_owned());
+        let hex_bytes = match hex_decode_result {
+            Ok(val) => {
+                val
+            },
+            Err(err) => {
+                if cmd.hex {
+                    // only raise hex decoding error if hex flag is set
+                    return Err(err.into())
+                }
+                // return empty vec that will never be used, to satisfy match
+                vec![]
+            }
+        };
+
+        let msg_req = if cmd.hex {
+            req(&to, hex_bytes.as_slice())
+        } else {
+            req(&to, cmd.message.as_bytes())
+        };
+
+        rpc.request_with_timeout(msg_req, Duration::from_secs(cmd.timeout))
             .await?;
-        let res = rpc.parse_response::<Vec<u8>>()?;
-        println!(
-            "{}",
-            String::from_utf8(res).context("Received content is not a valid utf8 string")?
-        );
+        let res = {
+            let res = rpc.parse_response::<Vec<u8>>()?;
+            if cmd.hex {
+                hex::encode(res)
+            } else {
+                String::from_utf8(res).context("Received content is not a valid utf8 string")?
+            }
+        };
+        opts.terminal.write_line(&res)?;
```

Comment:
This is still not right, as this is logging a message to stderr instead of writing the command output to stdout. To do that, you have to do the following: `opts.terminal.stdout().plain(res).write_line()?;`. Although you'll have to do a slight modification to the `go` function signature to receive the full `CommandGlobalOpts` instance (or flatten out the function code within the `rpc` function, up to you).

Reply:
Modified the go signature and fixed the logging call.